bitcoin/src/blockdata/transaction.rs

```diff
@@ -1063,17 +1080,18 @@ impl InputWeightPrediction {
         T::Item: Borrow<usize>,
     {
         let (count, total_size) = witness_element_lengths.into_iter().fold(
-            (0usize, 0),
+            (0usize, 0u32),
             |(count, total_size), elem_len| {
                 let elem_len = *elem_len.borrow();
-                let elem_size = elem_len + compact_size::encoded_size(elem_len);
+                let elem_size = Self::saturate_to_u32(elem_len) + Self::encoded_size(elem_len);
                 (count + 1, total_size + elem_size)
             },
         );
-        let witness_size =
-            if count > 0 { total_size + compact_size::encoded_size(count) } else { 0 };
+        let witness_size = if count > 0 { total_size + Self::encoded_size(count) } else { 0 };
         let script_size = input_script_len + compact_size::encoded_size(input_script_len);
 
+        let script_size = script_size as usize; // cast ok, u32 fits supported usize.
+        let witness_size = witness_size as usize; // cast ok, u32 fits supported usize.
```

Comment:
In 1f5a4b0a44dbe92b30b48678583aed9223de5da9:

We should just use `u32` internally. Then we don't need to make assumptions about whether `usize` might be smaller than `u32`.

Reply:
Good idea.  Done.